enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,458 @@
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoveryStatus struct {
+	Phase string `json:"phase,omitempty"`
+	// RelatedObjects is a list of objects created and maintained by this
+	// operator. Object references will be added to this list after they have
+	// been created AND found in the cluster.
+	// +optional
+	RelatedObjects []corev1.ObjectReference `json:"relatedObjects,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeNodeDiscovery `json:"items"`
+}
+
+type LocalVolumeNodeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              NodeDeviceMapSpec   `json:"spec"`
+	Status            NodeDeviceMapStatus `json:"status,omitempty"`
+}
+
+type NodeDeviceMapSpec struct {
+	NodeName string `json:"nodeName"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type NodeDeviceMapStatus struct {
+	Phase      string `json:"phase,omitempty"`
+	TimeStramp string `json:"timestramp,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
+	Ro         string   `json:"ro,omitempty"`
+	Parttype   string   `json:"parttype,omitempty"`
+	Serial     string   `json:"serial,omitempty"`
+	Size       string   `json:"size,omitempty"`
+	Mountpoint string   `json:"mountpoint,omitempty"`
+	Rm         string   `json:"rm,omitempty"`
+	Fstype     string   `json:"fstype,omitempty"`
+	Children   []Device `json:children,omitempty`
+}
+```
+
+### Workflow of `Auto Device Discovery`
+The discovery of local devices will be done by the `LocalVolumeDiscovery` and `LocalVolumeNodeDiscovery` CR. 
+#### LocalVolumeDiscovery CR
+- Once the `LocalVolumeDiscovery` CR is deployed, the controller of the `LocalVolumeDiscovery` CR will be managing `LocalVolumeNodeDiscovery` CR for each node that matches the `nodeSelector`.
```

Comment:
All the controller for LocalVolumeDiscovery does is generate the LocalVolumeNodeDiscovery CRs, right? "Managing" seems like too strong of a term since there is no other action than to generate the other CRs.

Reply:
replaced `managed` with `generating`.